### PR TITLE
Feature/billed items

### PR DESCRIPTION
**Pull request by olf42**
Billed invoices can be created with

$ rechnung bill-items 2019 10
$ rechnung bill-items 2019 11
$ rechnung bill-items 2019 12

which will create items to be billed for all contracts,
which are active in the given month. Next:

$ rechnung create-billed-invoices 2019.Q4

will create invoices from the billed items. All invoices
will get the suffix 2019.Q4.

Unfortunately this also required a couple of other changes

* contract_data is now passed plain into template. Elements
can now be accessed directly, without "contract."
* invoice_data is now passed plain into template. Elements
can now be accessed directly, withoud "invoice."
* test_fixtures assets folder is symlink to the orig folder,
as we should verify, that the assets we ship work with our tests..

### CODE REVIEW

**Comment by olf42** [comment]
TODO: Improve Testcoverage and fix circleci

**Comment by olf42** [comment]
All Done. Ready for review.

File: rechnung/cli.py [hunk 1]
```diff
@@ -32,6 +32,38 @@ def init():
...
+@click.option("-c", "--cid-only", help="One customer only.")
+@click.option(
+    "-d", "--dry", is_flag=True, default=False, help="Don't write the changes."
+)
+def bill_items(year, month, cid_only, dry):
+    """
+    Bill items for all active contracts (or just one with --cid-only).
+
+    The items will be added to a list, and put into an invoice, the 
+    next time create-invoices is run.
```
Comment (aparcar) [inline rechnung/cli.py:47]: Add params to docstring

File: rechnung/cli.py [hunk 2]
```diff
@@ -124,6 +156,19 @@ def send_invoices(year, month, cid_only=None, force=False):
     settings = get_settings_from_cwd(cwd)
     invoice.send_invoices(settings, year, month, cid_only, force)
 
+@cli1.command()
+@click.argument("suffix")
+@click.option("-c", "--cid_only")
+@click.option("-f", "--force-resend", "force", is_flag=True)
+def send_invoices_suffix(suffix, cid_only=None, force=False):
+    """
+    Send invoices by email (selected by suffix instead of year and month).
```
Comment (aparcar) [inline rechnung/cli.py:165]: Add params to docstring

File: rechnung/cli.py [hunk 3]
```diff
@@ -32,6 +32,38 @@ def init():
...
+    invoice.bill_items(settings, year, month, cid_only, dry)
+
+
+@cli1.command()
+@click.argument("suffix")
+@click.option("-c", "--cid-only")
+@click.option("-f", "--force-recreate", "force", is_flag=True)
+def create_billed_invoices(suffix, cid_only=None, force=False):
+    """
+    Mass create invoices from billed items.
```
Comment (aparcar) [inline rechnung/cli.py:60]: Add params to docstring

File: rechnung/contract.py [hunk 4]
```diff
@@ -1,3 +1,4 @@
+import arrow
```
Comment (aparcar) [inline rechnung/contract.py:1]: Do we really need arrow and can't stay with the standard datetime lib?

**Review by aparcar — APPROVED** [review]
Overall great. I'm just a bit scared of increasing filesizes and would move things at some point to a database, like peewee. But this is a future discussion...

File: rechnung/contract.py (hunk 4 above)
Comment (olf42) [reply rechnung/contract.py:1]: Arrow provides lots of functionality (i.e. easy to use ceil() and floor() functions, timedeltas, etc. as well as string-parsing without specifying the format first. I strongly suggest to continue using it, and even replace datetime() completely.

**Comment by olf42** [comment]
Doing more work in an (in memory) database to make the code clearer and possibly reusable definitely is on my roadmap for 2020. I already did an PoC with that (especially saving the changes made in peewee using the marshmallow serialization library). I will add the params to the docstrings and then merge this PR. Thank you for the review.

File: rechnung/cli.py (hunk 3 above)
Comment (olf42) [reply rechnung/cli.py:60]: Doesn't make sense at this point, as the docstring is display with --help to the user. Help-Texts is the way to go for cli-functions.

File: rechnung/cli.py (hunk 2 above)
Comment (olf42) [reply rechnung/cli.py:165]: Doesn't make sense at this point, as the docstring is display with --help to the user. Help-Texts is the way to go for cli-functions.

File: rechnung/cli.py (hunk 1 above)
Comment (olf42) [reply rechnung/cli.py:47]: Doesn't make sense at this point, as the docstring is display with --help to the user. Help-Texts is the way to go for cli-functions.

**Pull request merged**